### PR TITLE
Docs: Update README with ESM instructions (fixes #474)

### LINKED ISSUE #474: ESM API docs
We should update README with the new ESM API, once https://github.com/eslint/espree/pull/469 gets merged.

### PR DESCRIPTION
**Pull request by nzakas**
Updates the README with ESM instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +89,8 @@ Even if `options` is empty or undefined or `options.tokens` is `false`, it assig
 **Example :**
 
 ```js
+import espree from "espree";
+
```
Comment (aladdin-add) [inline README.md:93]: it should be `import * as espree from "espree"` - as espree do not have `export defautl`.

or just `import {tokenize} from "espree"`

File: docs/README.md [hunk 2]
```diff
@@ -81,6 +89,8 @@ Even if `options` is empty or undefined or `options.tokens` is `false`, it assig
 **Example :**
 
 ```js
+import espree from "espree";
+
```
Comment (aladdin-add) [inline docs/README.md:93]: same here. :)

File: README.md [hunk 3]
```diff
@@ -81,6 +89,8 @@ Even if `options` is empty or undefined or `options.tokens` is `false`, it assig
 **Example :**
 
 ```js
+import espree from "espree";
```
Comment (nzakas) [inline README.md:92]:
```suggestion
import * as espree from "espree";
```

File: docs/README.md [hunk 4]
```diff
@@ -81,6 +89,8 @@ Even if `options` is empty or undefined or `options.tokens` is `false`, it assig
 **Example :**
 
 ```js
+import espree from "espree";
```
Comment (nzakas) [inline docs/README.md:92]:
```suggestion
import * as espree from "espree";
```

File: docs/README.md (hunk 2 above)
Comment (nzakas) [reply docs/README.md:93]: Got it, thanks

**Comment by nzakas** [comment]
Should be all set now. :+1:

**Review by mdjermanovic — APPROVED** [review]

**Pull request merged**